gobblin-service/src/main/java/org/apache/gobblin/service/modules/spec/JobExecutionPlan.java

```diff
@@ -112,10 +112,10 @@ private static JobSpec buildJobSpec(FlowSpec flowSpec, Config jobConfig, Long fl
       // job names are assumed to be unique within a dag.
       int hash = flowInputPath.hashCode();
       jobName = Joiner.on(JOB_NAME_COMPONENT_SEPARATION_CHAR).join(flowGroup, flowName, jobName, edgeId, hash);
-      // jobNames are commonly used as a directory name, which is limited to 255 characters
+      // jobNames are commonly used as a directory name, which is limited to 255 characters (account for potential prefixes added/file name lengths)
```

Comment:
Should we add flowExecutionId before calculating hash?

Reply:
1. No because then it would create a different work directory per execution which would lead to jars not being cached etc.
2. Not sure but I believe it was decided a while back that it would be more unique in the multihop/parallel job setting than edgeId or flowName, as you can have one edge map to multiple jobs.